Sources/Networking/Core/DownloadAPIManaging.swift

```diff
@@ -41,13 +41,52 @@ public protocol DownloadAPIManaging {
     func progressStream(for task: URLSessionTask) -> AsyncStream<URLSessionTask.DownloadState>
 }
 
-// MARK: - Provide request with default nil resumable data, retry configuration
 public extension DownloadAPIManaging {
+    /// Initiates a download request for a given fileURL, with optional resumable data and retry configuration.
+    /// - Parameters:
+    ///   - fileURL: A URL of a file which will be downloaded.
+    ///   - resumableData: Optional data the download request will be resumed with.
+    ///   - retryConfiguration: Configuration for retrying behaviour.
+    /// - Returns: A download result consisting of `URLSessionDownloadTask` and `Response`
+    func downloadRequest(
+        _ fileURL: URL,
+        resumableData: Data? = nil,
+        retryConfiguration: RetryConfiguration? = .default
+    ) async throws -> DownloadResult {
+        try await downloadRequest(
+            DownloadRouter(fileURL: fileURL),
+            resumableData: resumableData,
+            retryConfiguration: retryConfiguration
+        )
+    }
+
+    // Provide request with default nil resumable data, retry configuration
     func downloadRequest(
         _ endpoint: Requestable,
         resumableData: Data? = nil,
         retryConfiguration: RetryConfiguration? = .default
     ) async throws -> DownloadResult {
-        try await downloadRequest(endpoint, resumableData: resumableData, retryConfiguration: retryConfiguration)
+        try await downloadRequest(
+            endpoint,
+            resumableData: resumableData,
+            retryConfiguration: retryConfiguration
+        )
+    }
+}
+
+/// A Router used for basic use case of downloading a file from a URL.
```

Comment:
I'm not sure if I like having this private struct in the same file as is the public interface to the library 🤔 . Wouldn't it be better somewhere else (own file?)

Reply:
Sure I can do that if you want. I just like to put simple single use types into the source file where they are used in, since they don't need to be exposed 🤷‍♂️